src/ports/catalog/queries.ts

```diff
@@ -237,8 +241,8 @@ export const getIsSoldOutWhere = () => {
 
 export const getIsOnSale = (filters: CatalogFilters) => {
   return filters.isOnSale
-    ? SQL`((search_is_store_minter = true AND available > 0) OR listings_count IS NOT NULL)`
-    : SQL`((search_is_store_minter = false OR available = 0) AND listings_count IS NULL)`
+    ? SQL`((search_is_store_minter = true AND available > 0) OR (nfts_with_orders.orders_listings_count IS NOT NULL OR offchain_orders.count IS NOT NULL))`
+    : SQL`((search_is_store_minter = false OR available = 0) AND (nfts_with_orders.orders_listings_count IS NULL AND offchain_orders.count IS NULL))`
```

Comment:
This is the same behavior as before, but does it make sense to filter only the "not on sale"? shouldn't it be if `isOnSale` is undefined for example that it shows all items?